src/progress-steps/step.js

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright (c) 2018 Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+
+// @flow
+
+import * as React from 'react';
+import {getOverrides} from '../helpers/overrides';
+import {
+  StyledStep,
+  StyledIcon,
+  StyledInnerIcon,
+  StyledContent,
+  StyledContentTitle,
+  StyledContentTail,
+  StyledContentDescription,
+} from './styled-components';
+
+import type {StepPropsT} from './types';
+
+function Step({
+  overrides = {},
+  isCompleted,
+  isActive,
+  isLast,
+  title,
+  children,
+}: StepPropsT) {
+  const [Root, rootProps] = getOverrides(overrides.Root, StyledStep);
+  const [Icon, iconProps] = getOverrides(overrides.Icon, StyledIcon);
+  const [InnerIcon, innerIconProps] = getOverrides(
+    overrides.InnerIcon,
+    StyledInnerIcon,
+  );
+  const [Tail, tailProps] = getOverrides(overrides.Tail, StyledContentTail);
+  const [Content, contentProps] = getOverrides(
+    overrides.Content,
+    StyledContent,
+  );
+  const [Title, titleProps] = getOverrides(overrides.Title, StyledContentTitle);
+  const [Description, descriptionProps] = getOverrides(
+    overrides.Description,
+    StyledContentDescription,
+  );
+
+  return (
+    <Root {...rootProps}>
```

Comment:
Same comment here and the rest of elements below regarding passing the shared props to every single underlying element